packages/components/data-table/src/data-row.js

```diff
@@ -15,10 +15,18 @@ const DataRow = (props) => {
   const handleRowCollapseClick = () => {
     collapseRow(!isRowCollapsed);
   };
+
+  // update the collapsed state if isTruncated options are changed for the whole row
+  React.useEffect(() => {
+    if (rowHasTruncatedColumn) {
+      collapseRow(true);
+    } else if (!rowHasTruncatedColumn) {
+      collapseRow(false);
+    }
+  }, [rowHasTruncatedColumn]);
+
   const shouldRenderCollapseButton = (totalColumnsLength, currentColumnIndex) =>
-    totalColumnsLength - 1 === currentColumnIndex &&
-    ((isRowCollapsed && rowHasTruncatedColumn) ||
-      (rowHasTruncatedColumn && !isRowCollapsed));
+    totalColumnsLength - 1 === currentColumnIndex && isRowCollapsed;
```

Comment:
This is unrelated to the PR but I felt really dumb when I saw that we had this in the code 😆